src/setfit/trainer.py

```diff
@@ -119,7 +119,11 @@ def train(self):
             logger.info("Applying column mapping to training dataset")
             self.train_dataset = self._apply_column_mapping(self.train_dataset, self.column_mapping)
         x_train = self.train_dataset["text"]
-        y_train = self.train_dataset["label"]
+
+        if self.model.multi_target_strategy:
+            y_train = self.train_dataset.remove_columns("text").to_pandas().values
```

Comment:
I think this won't work for generic datasets where one has more than two columns. I know this logic comes from the multilabel script, but I wonder if we can make use of the column mapping added in #75 and always derive `x_train` and `y_train` from `text` and `label` columns?

In other words, can we drop this if/else statement alltogether?

Reply:
I have a question here: 

I like the idea of expecting always `text` and `label` columns also for multilabel case, but when using datasets like `lex_glue` (`eurlex`) labels are non-binary sequences (e.g. `[ 28, 32, 33, 91, 96, 97 ]`) while `sentence_pairs_generation_multilabel` expects binary sequences. How do you suggest to handle them?

Reply:
The huggingface libraries typically leave binarization to the user (using `map` and the like), and handling binarization in a generic way can get [rather involved](https://github.com/chatdesk/grouphug/blob/master/grouphug/dataset_formatter.py#L274).
Could a reasonable intermediate be to allow a list for the `label` value in columns mapping? That way you can handle `go_emotions` datasets at least somewhat cleanly.